github/GithubIntegration.py

```diff
@@ -0,0 +1,198 @@
+            app_id=None,
+            app_private_key=None,
+            app_installation_id=None,
+            app_token_permissions=None,
+            base_url=self.base_url,
+            timeout=Consts.DEFAULT_TIMEOUT,
+            user_agent="PyGithub/Python",
+            per_page=Consts.DEFAULT_PER_PAGE,
+            verify=True,
+            retry=None,
+            pool_size=None,
+        )
+
+    def _get_headers(self):
+        """
+        Get headers for the requests.
+
+        :return: dict
+        """
+        return {
+            "Authorization": f"Bearer {self.create_jwt()}",
+            "Accept": Consts.mediaTypeIntegrationPreview,
+            "User-Agent": "PyGithub/Python",
+        }
+
+    def _get_installed_app(self, url):
+        """
+        Get installation for the given URL.
+
+        :param url: str
+        :rtype: :class:`github.Installation.Installation`
+        """
+        try:
+            headers, response = self.__requester.requestJsonAndCheck(
+                "GET", url, headers=self._get_headers()
+            )
+        except UnknownObjectException:
+            raise
+        except GithubException:
+            raise
+        return Installation(
+            requester=self.__requester,
+            headers=headers,
+            attributes=response,
+            completed=True,
+        )
+
+    def create_jwt(self):
+        """
+        Creates a signed JWT, valid for 60 seconds by default. Could be set to a maximum of 600 seconds.
+        https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-a-github-app
+
+        :return string:
+        """
+        now = int(time.time())
+        payload = {"iat": now, "exp": now + self.jwt_expiry, "iss": self.integration_id}
+        encrypted = jwt.encode(payload, key=self.private_key, algorithm="RS256")
+
+        if isinstance(encrypted, bytes):
+            encrypted = encrypted.decode("utf-8")
+
+        return encrypted
+
+    def get_access_token(self, installation_id, permissions=None):
+        """
+        :calls: `POST /app/installations/{installation_id}/access_tokens <https://docs.github.com/en/rest/apps/apps#create-an-installation-access-token-for-an-app>`
+        :param installation_id: int
+        :param permissions: dict
+        :return: :class:`github.InstallationAuthorization.InstallationAuthorization`
+        """
+        if permissions is None:
+            permissions = {}
+
+        if not isinstance(permissions, dict):
+            raise GithubException(
+                status=400, data={"message": "Invalid permissions"}, headers=None
+            )
+
+        body = {"permissions": permissions}
+        try:
+            headers, response = self.__requester.requestJsonAndCheck(
+                "POST",
+                f"/app/installations/{installation_id}/access_tokens",
+                input=body,
+            )
+        except UnknownObjectException:
```

Comment:
Why are you catching and then raising these two exceptions here?

Reply:
I honestly don't remember, but I see other code that does this, maybe it is to alert readers as to the exceptions that can be raised by this method?

This could also be achieved by using `:raises UnknownObjectException:` and `:raises GithubException:` in the docstring. Do you prefer I change it?

Reply:
We commonly catch the exceptions to change the return type -- for example `Team.get_repo_permission`, so I don't see any value for us catching the exceptions and then re-raising them. We may want to document in the docstrings which exceptions can be raised, but I'd prefer we punt that to the future.

Reply:
I removed the useless exception handling.